docs/content/preview/releases/techadvisories/ta-23476.md

```diff
@@ -0,0 +1,169 @@
+
+When these non-zero microsecond timestamp values are stored, the microsecond information will be lost when the precision is adjusted to milliseconds for the wire protocol (for example, when responding to a user query). As a result, you will always see timestamp values with only millisecond precision.
+
+This can create issues in certain scenarios as follows:
+
+- Equality conditions: If you query for a timestamp value that was just read, the condition will not match, as the microsecond component will differ (0 for the query, non-zero for the stored value).
+
+- Index scans: If the primary key of the main table includes an affected timestamp column, the data in both the table and the index will be in microsecond precision, so they will match. However, when performing an index scan, the main-table read request uses the same protocol as the YCQL API, which adjusts the precision to milliseconds. This may cause the main-table read request to not match, and the index scan result to miss the affected rows with non-zero microsecond values.
+
+## Examples
+
+### Primary-key scan with equality condition
+
+1. Set up the keyspace and table as follows:
+
+    ```cql
+    CREATE KEYSPACE temp;
+    USE temp;
+    CREATE TABLE test(id int, ts timestamp, v1 int, v2 int, primary key( id, ts)) WITH transactions = {'enabled': 'true'};
+    ```
+
+1. Insert a row using `currenttimestamp()` as follows:
+
+    ```cql
+    INSERT INTO test(id, ts, v1, v2) VALUES (1, currenttimestamp(), 5, 10);
+    SELECT * FROM test;
+    ```
+
+    ```output
+     id | ts                              | v1 | v2
+    ----+---------------------------------+----+----
+      1 | 2024-08-21 17:44:47.647000+0000 |  5 | 10
+    ```
+
+    As expected the row is present. However, most likely the internal row has a non-zero microsecond component, just that due to the YCQL API millisecond standard, it will always return `000` for the microsecond part.
+
+1. Select the row using WHERE condition as follows:
+
+    ```cql
+    SELECT * FROM test WHERE id = 1 AND ts = '2024-08-21 17:44:47.647000+0000';
+    ```
+
+    ```output
+     id | ts | v1 | v2
+    ----+----+----+----
+    ```
+
+    ```cql
+    SELECT * FROM test WHERE id = 1 AND ts >= '2024-08-21 17:44:47.647000+0000' and ts < '2024-08-21 17:44:47.648000+0000';
+    ```
+
+    ```output
+      id | ts                              | v1 | v2
+     ----+---------------------------------+----+----
+       1 | 2024-08-21 17:44:47.647000+0000 |  5 | 10
+    ```
+
+    A simple query with equality on the same (id and) timestamp value returned before does not match. However, as you can see changing this to a range query search for all timestamps within that millisecond does return the row, confirming that the row is there but the microsecond value is non-zero. Note that for the second query the upper bound is the next millisecond value (648 rather than 647) so it will match everything in [647000, 647999] range from microsecond perspective.
```

Comment:
```suggestion
    A simple query with equality on the same (id and) timestamp value returned before does not match. However, changing this to a range query search for all timestamps within that millisecond does return the row, confirming that the row is there but the microsecond value is non-zero. Note that for the second query the upper bound is the next millisecond value (648 rather than 647) so it will match everything in [647000, 647999] range from the microsecond perspective.
```